tests/test_ignore_names.py

```diff
@@ -0,0 +1,71 @@
+import pytest
+
+from vulture import core
+
+from . import check
+
+
+@pytest.fixture
+def check_ignore():
+    def examine(code, ignore_names, expected):
+        v = core.Vulture(verbose=True, ignore_names=ignore_names)
+        v.scan(code)
+        check(v.get_unused_code(), expected)
+    return examine
+
+
+def test_var(check_ignore):
+    code = """\
+foo = 1
+bar = 2
+three = 3
+funny = True
+"""
+    check_ignore(code, ['f*', 'bar'], ['three'])
+
+
+def test_function(check_ignore):
+    code = """\
+def foo_one():
+    pass
+def foo_two():
+    pass
+def foo():
+    pass
+def bar():
+    pass
+"""
+    check_ignore(code, ['foo*'], ['bar'])
+
+
+def test_class(check_ignore):
+    code = """\
+class Foo:
+    def __init__(self):
+        pass
+"""
+    check_ignore(code, ['Foo'], [])
+
+
+def test_property(check_ignore):
+    code = """\
+class Foo:
+    @property
+    def some_property(self, a):
+        return a
+
+    @property
+    def foo_bar(self):
+        return 'bar'
+"""
+    check_ignore(code, ['Foo', 'property'], ['some_property', 'foo_bar'])
+
+
+def test_attribute(check_ignore):
+    code = """\
+class Foo:
+    def __init__(self, attr_foo, attr_bar):
+        self.attr_foo = attr_foo
+        self.attr_bar = attr_bar
+"""
+    check_ignore(code, ['*_foo'], ['Foo', 'attr_bar'])
```

Comment:
We also need tests for decorator names and imports and tests for ``ba[rz]`` and ``f?o``. Also, we need to check that the matching is case-sensitive (ignoring foo doesn't ignore Foo).